templates/connection_init.erb

```diff
@@ -32,18 +39,33 @@ piddir=`dirname "${pidfile}"`
 user=`grep "setuid[[:space:]]*=" /etc/stunnel/stunnel.conf | sed -e 's/setuid[[:space:]]*=[[:space:]]*\(.*\)[[:space:]]*$/\1/'`;
 group=`grep "setgid[[:space:]]*=" /etc/stunnel/stunnel.conf | sed -e 's/setgid[[:space:]]*=[[:space:]]*\(.*\)[[:space:]]*$/\1/'`;
 
-# Set the SELinux Contexts Properly
-if [ -z "${pidfile}" ]; then
-    if [ ! -d "${piddir}" ]; then
-      mkdir -p --context=system_u:object_r:stunnel_var_run_t:s0 "${piddir}"
-      chown ${user}:${group} "${piddir}"
-    fi
-fi
-if [ -n $chroot ]; then
-    mkdir -p --context=system_u:object_r:stunnel_var_run_t:s0 "${chroot}${piddir}"
-    chown ${user}:${group} "${chroot}${piddir}"
-    pidfile="${chroot}${pidfile}";
+# # Set the SELinux Contexts Properly
+# if [ -z "${pidfile}" ]; then
+#     if [ ! -d "${piddir}" ]; then
+#       mkdir -p --context=system_u:object_r:stunnel_var_run_t:s0 "${piddir}"
+#       chown ${user}:${group} "${piddir}"
+#     fi
+# fi
+# if [ -n $chroot ]; then
+#     mkdir -p --context=system_u:object_r:stunnel_var_run_t:s0 "${chroot}${piddir}"
+#     chown ${user}:${group} "${chroot}${piddir}"
+#     pidfile="${chroot}${pidfile}";
+# fi
+
+<% if @_chroot.to_s.length > 0 -%>
+pidfile="<%= "#{@_chroot}#{@pid}" %>";
+piddir=`dirname "${pidfile}"`
+if [ ! -d "${piddir}" ]; then
+<%   if @_se_enabled -%>
+  mkdir -p --context=system_u:object_r:stunnel_var_run_t:s0 "<%= $piddir %>"
```

Comment:
See config.pp lines 135-140.  Looks like will only get here if using selinux and if selinux_current_mode fact is disabled.  Why would we want to set the contexts in this case?